### PR TITLE
Add support for build and push docker image

### PR DESCRIPTION
**Pull request by eddumelendez**
Resolves spring-cloud/spring-cloud-skipper#148
Resolves spring-cloud/spring-cloud-skipper#149

### CODE REVIEW

**Comment by markpollack** [comment]
Thanks so much!

File: spring-cloud-skipper-server/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FROM openjdk:8u141
```
Comment (trisberg) [inline spring-cloud-skipper-server/Dockerfile]: We use `java:8-alpine` for Data Flow images

File: spring-cloud-skipper-server/pom.xml [hunk 2]
```diff
@@ -11,6 +11,10 @@
         <version>1.0.0.BUILD-SNAPSHOT</version>
     </parent>
 
+    <properties>
+        <docker.image.prefix>spring-io</docker.image.prefix>
+    </properties>
```
Comment (trisberg) [inline spring-cloud-skipper-server/pom.xml]: could we use `springcloud` here, that's what we use for Data Flow

File: spring-cloud-skipper-server/pom.xml [hunk 3]
```diff
@@ -53,4 +57,43 @@
             </plugin>
         </plugins>
     </build>
+
+	<profiles>
+		<profile>
+			<id>docker</id>
+
```
Comment (trisberg) [inline spring-cloud-skipper-server/pom.xml]: do we want this as a profile or just use `./mvnw package dockerfile:build` without adding the plugin to the package phase

**Comment by eddumelendez** [comment]
@trisberg thanks for feedback, I have update the docker image and the prefix in the new commit. Regarding to the docker profile I will wait until the decision is taken.

File: spring-cloud-skipper-server/pom.xml (hunk 3 above)
Comment (sabbyanandan) [reply spring-cloud-skipper-server/pom.xml]: Probably just fine with `dockerfile:build` just to be consistent with other projects that generate docker images.

**Comment by sabbyanandan** [comment]
Thanks for the update, @eddumelendez! See my comment re: profile ^^. Hopefully, when you're done with that change, we can go ahead and merge it. 😄

**Comment by eddumelendez** [comment]
change is done 😃

**Comment by trisberg** [comment]
LGTM

**Comment by trisberg** [comment]
Rebased and merged as commit e5c52a2ee715a2eb75baa83f4e11d228ecb98d39

**Pull request closed without merging**